chain/network/src/routing_table_actor.rs

```diff
@@ -129,25 +486,32 @@ impl Handler<RoutingTableMessages> for RoutingTableActor {
     fn handle(&mut self, msg: RoutingTableMessages, _ctx: &mut Self::Context) -> Self::Result {
         match msg {
             RoutingTableMessages::AddEdges(edges) => {
-                for edge in edges.iter() {
-                    #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
-                    {
-                        let se = edge.to_simple_edge();
-                        self.peer_ibf_set.add_edge(&se);
-                    }
-                    self.edges_info.insert((edge.peer0.clone(), edge.peer1.clone()), edge.clone());
+                for edge in edges {
+                    self.add_verified_edge(edge);
                 }
                 RoutingTableMessagesResponse::Empty
             }
             RoutingTableMessages::RemoveEdges(edges) => {
-                for edge in edges.iter() {
-                    #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
-                    self.peer_ibf_set.remove_edge(&edge.to_simple_edge());
-
-                    self.edges_info.remove(&(edge.peer0.clone(), edge.peer1.clone()));
-                }
+                self.remove_edges(&edges);
                 RoutingTableMessagesResponse::Empty
             }
+            RoutingTableMessages::AddVerifiedEdges { edges } => {
+                RoutingTableMessagesResponse::AddVerifiedEdgesResponse(
+                    self.add_verified_edges_to_routing_table(edges),
+                )
+            }
+            RoutingTableMessages::RoutingTableUpdate { prune, prune_edges_after } => {
+                let edges_to_remove = if self.needs_routing_table_recalculation {
+                    self.recalculate_routing_table(prune, prune_edges_after)
+                } else {
+                    Vec::new()
+                };
+                self.needs_routing_table_recalculation = false;
+                RoutingTableMessagesResponse::RoutingTableUpdateResponse {
+                    edges_to_remove,
```

Comment:
as these are 'edges_TO_remove', who is doing the 'actual removing'?

as I see that we're leaving only AdvRemoveEdges as a way to remove stuff. What am I missing?

Reply:
I renamed the above function name to `recalculate_routing_table_and_maybe_prune_edges`.

The only way to remove edges from routing table is by pruning, which is only done while doing routing table update.
Edge pruning used to be done by `PeerManager`, so it was telling `RoutingTableActor` to remove edges. Now, the situation is reversed.